src/database/index.tsx

```diff
@@ -349,13 +350,34 @@ export const getFacilityModelVersions = async ({
     let modelVersions = historyResults.docs.map((doc) =>
       buildModelInputs(doc.data()),
     );
-    return distinctByObservedAt
-      ? uniqBy(
+    if (distinctByObservedAt) {
+      // if observedAt has not been normalized,
+      // we have to re-sort in memory to properly group by observedAt
+      let normalized = !modelVersions.some((version, index) => {
+        if (!index) return;
+
+        const { observedAt } = version;
+        const { observedAt: prevObservedAt } = modelVersions[index - 1];
+        return (
+          isSameDay(observedAt, prevObservedAt) &&
+          !isEqual(observedAt, prevObservedAt)
+        );
+      });
+      if (!normalized) {
+        modelVersions = orderBy(
           modelVersions,
-          // make sure time doesn't enter into the uniqueness
-          (record) => record.observedAt.toDateString(),
-        )
-      : modelVersions;
+          [(version) => version.observedAt.toDateString(), "updatedAt"],
```

Comment:
Is this sorting first by observedAt and then secondarily by updatedAt? Ascending by observedAt and then descending by updatedAt? I might not be thinking clearly but wouldn't we want the two sorts to be in the same direction?

Reply:
yeah it's a quirky thing (you will see we are doing the same thing in the query) ... the reason is that `uniqBy` returns the first item in array order that matches its predicate, and that's what we're using to filter duplicate days

Reply:
Ah, I understand. Might make sense to add a comment explaining that but I don't feel strongly towards it. It's not unintuitive, it just relies on a bit more context to get.